test_src/TableTest.cpp

```diff
@@ -0,0 +1,542 @@
+
+            assert (tableItr.hasNext());    // 3rd row
+            rowFetched = tableItr.next();
+            CPPUNIT_ASSERT(rowFetched.columnCount() == columns.size());
+            schemaRead = rowFetched.columns();
+            CPPUNIT_ASSERT(schemaRead == columns);
+            verifyNullValues(rowFetched);
+
+            assert (!tableItr.hasNext());   // no more rows left
+
+            // serialization
+            int32_t tableSerializeSize = table.getSerializedSize();
+            char *data = new char[tableSerializeSize];
+            boost::scoped_ptr<char> scopedData(data);
+            ByteBuffer tableBB(scopedData.get(), tableSerializeSize);
+            int32_t serializedSize = table.serializeTo(tableBB);
+            if (tableSerializeSize != serializedSize) {
+                std::cout << "expected serialized size: " << tableSerializeSize <<" got: " << serializedSize <<std::endl;
+            }
+            CPPUNIT_ASSERT(serializedSize == tableSerializeSize);
+
+            // verify the table size from serialized bb
+            int32_t tableSize = tableBB.getInt32(0);
+            CPPUNIT_ASSERT((serializedSize - 4) == tableSize);
+
+        }
+
+        void testRowBuilderRudimentary() {
+            std::vector<voltdb::Column> columns;
+            allTypeColumns(columns);
+
+            fillRandomValues();
+            RowBuilder row(columns);
+            int32_t rowDataSize = populateRow(row, false);
+
+            int32_t serializedSize = row.getSerializedSize();
+            if ((rowDataSize + 4) != serializedSize) {
+                std::cout << __LINE__ << (rowDataSize+4) << " : " << serializedSize << std::endl;
+            }
+            CPPUNIT_ASSERT((rowDataSize + 4) == serializedSize);
+
+            char* data = (char *) malloc(serializedSize);
+            SharedByteBuffer byteBuffer(data, serializedSize);
+
+            byteBuffer.ensureRemaining(serializedSize);
+            rowDataSize = row.serializeTo(byteBuffer);
+            CPPUNIT_ASSERT(rowDataSize == serializedSize);
+
+            // populate with nulls
+            rowDataSize = populateRow(row, true);
+            serializedSize = row.getSerializedSize();
+            if ((rowDataSize + 4) != serializedSize) {
+                std::cout << __LINE__ << (rowDataSize+4) << " : " << serializedSize << std::endl;
+            }
+            CPPUNIT_ASSERT((rowDataSize + 4) == serializedSize);
+
+            byteBuffer.clear();
+            byteBuffer.ensureRemaining(serializedSize);
+            rowDataSize = row.serializeTo(byteBuffer);
+            CPPUNIT_ASSERT(rowDataSize == serializedSize);
+
+            // verify the row size from serialized bb
+            int32_t rowSize = byteBuffer.getInt32(0);
+            CPPUNIT_ASSERT((serializedSize - 4) == rowSize);
+        }
+
+        void testTableConstructWitnNoColumns() {
+            std::vector<voltdb::Column> columns;
+            Table table (columns);
```

Comment:
Currently this test generates an exception as empty column table is not allowed. I will update it so that can assert on message.